Functions/xASL_quant_SliceTiming.m

```diff
@@ -111,6 +112,43 @@
 	SliceTiming = (0:1:(nSlices-1)) * x.Q.SliceReadoutTime;
 elseif length(x.Q.SliceReadoutTime) == nSlices
 	SliceTiming = x.Q.SliceReadoutTime;
+	
+	% Check for SliceTiming consistency between slices if a vector was provided
+	if nSlices > 1
+		SliceTimingDiff = SliceTiming(2:end) - SliceTiming(1:(end-1));
+		% See if the slice-timing difference between slices is equal for all slices
+		[SliceTimingDiffUnique,SliceTimingDiffIndex] = uniquetol(SliceTimingDiff,0.01);
+		
+		% Make sure that the positive values are listed first
+		[SliceTimingDiffUnique,SliceTimingDiffIndexSort] = sort(SliceTimingDiffUnique,'descend');
+		SliceTimingDiffIndex = SliceTimingDiffIndex(SliceTimingDiffIndexSort);
+		
+		if length(SliceTimingDiffUnique) > 2
+			fprintf('Warning: Inconsistent SliceTiming between slices\n');
+		elseif length(SliceTimingDiffUnique) == 2
```

Comment:
Are you sure multiband can only have 2 values? I thought it could also have higher acceleration factors, leading to multiple SliceReadoutTimes

Reply:
Haha - yes and no. Multiband usually has higher acceleration factors, but we don't look here at the SliceTiming, but at SliceTimingDiff!!
So an example of multiband with factor 3, SliceTime 30, and 12 slices  would be
0 30 60 90 0 30 60 90 0 30 60 90
So it would create 3 blocks of 4 slices. So 4 different times. BUT the differences of the neighboring slices are only 2: 30 and -90.

So the '2' here doesn't refer to the number of blocks or acceleration factor, but to the fact that the SliceTime increases equally within block and decreases equally between blocks = 2 different changes.